addons/web/static/src/search/utils/dates.js

```diff
@@ -209,18 +210,21 @@ export function constructDateRange(params) {
         setParam.month = QUARTERS[setParam.quarter].coveredMonths[0];
         delete setParam.quarter;
     }
-    const date = referenceMoment.set(setParam).plus(plusParam || {});
+    const date = referenceMoment
+        .set(setParam)
+        .plus(plusParam || {})
+        .setZone("utc", { keepLocalTime: true });
     // compute domain
     let leftDate = date.startOf(granularity);
     let rightDate = date.endOf(granularity);
     let leftBound;
     let rightBound;
     if (fieldType === "date") {
-        leftBound = leftDate.toFormat("yyyy-MM-dd");
-        rightBound = rightDate.toFormat("yyyy-MM-dd");
+        leftBound = serializeDate(leftDate);
+        rightBound = serializeDate(rightDate);
     } else {
-        leftBound = leftDate.toUTC().toFormat("yyyy-MM-dd HH:mm:ss");
-        rightBound = rightDate.toUTC().toFormat("yyyy-MM-dd HH:mm:ss");
+        leftBound = serializeDateTime(leftDate);
+        rightBound = serializeDateTime(rightDate);
```

Comment:
I don't understand this...
leftDate and rightDate are moment objects, aren't they ?
but serializeDate[Time]() functions take luxon's DateTime objects...

Reply:
No they aren't, I now this is very confusing but `referenceMoment` is actually a luxon DateTime object

Reply:
This should not be done in this PR, but in master we should maybe rename this to avoid confusion